click/_bashcomplete.py

```diff
@@ -97,6 +96,23 @@ def is_incomplete_argument(current_params, cmd_param):
         return True
     return False
 
+def get_user_autocompletions(ctx, args, incomplete, cmd_param):
+    """
+    :param ctx: context associated with the parsed command
+    :param args: full list of args
+    :param incomplete: the incomplete text to autocomplete
+    :param cmd_param: command definition
+    :return: all the possible user-specified completions for the param
+    """
+    choices = []
+    if isinstance(cmd_param.type, Choice):
+        choices = cmd_param.type.choices
```

Comment:
You could just `return cmd_param.type.choices` here (same for the elif branch) and add an else branch that does `return []`. Then you don't have to create a list everytime.